payload_stage2/source/main.c

```diff
@@ -39,38 +39,87 @@ void main(void)
     mountSD();
 
     u32 payloadFound;
+    u32 defaultInit;
+    u32 skipInit = 0;
 
     if(fileRead((void *)PAYLOAD_ADDRESS, "homebrew/3ds/a9nc.bin")) // Full A9NC support
     {
         payloadFound = 1;
+        skipInit = 1;
         ownArm11(1);
         clearScreens();
         i2cWriteRegister(3, 0x22, 0x2A); //Turn on backlight
         f_unlink("homebrew/3ds/a9nc.bin");
     }
-    else if(fileRead((void *)PAYLOAD_ADDRESS, "homebrew/3ds/boot.bin"))
```

Comment:
I thought I'd added backwards compatibility. I seem to have misunderstood how it works anyway, it seems not to behave as expected.